Add Spice-Cache-Key docs to caching section (#1076)
spiceai/docs: website/docs/features/caching/index.md

# Skip cache for this search, but cache the results for future searches
spice search --cache-control no-cache
```

## Custom Cache Keys

> You may set the `Spice-Cache-Key` header to supply your own cache key. When set, a supplied cache key takes precedence over `caching.sql_results.cache_key_type`.

Reviewer: ```suggestion
Set the `Spice-Cache-Key` header to supply a custom cache key. When set, a supplied cache key takes precedence over `caching.sql_results.cache_key_type`.
```

+----+--------+-------+----------------+
```

To share a cache key for these queries, set `Spice-Cache-Key`. The first request is a cache miss:

```bash
$ curl -i -XPOST http://localhost:8090/v1/sql -H"spice-cache-key: users_spiceai" -d "select * from users where org_id = 1;"
HTTP/1.1 200 OK
content-type: application/json
x-cache: Miss from spiceai
results-cache-status: MISS
vary: Spice-Cache-Key
vary: origin, access-control-request-method, access-control-request-headers
content-length: 119
date: Thu, 24 Jul 2025 14:15:53 GMT

[{"id":1,"org_id":1,"name":"Jane","email":"jane@spice.ai"},{"id":2,"org_id":1,"name":"Sarah","email":"sarah@spice.ai"}]
```

The subsequent request with the different (but semantically equivalent) query is a cache hit:

```bash
$ curl -i -XPOST http://localhost:8090/v1/sql -H"spice-cache-key: users_spiceai" -d "select * from users where split_part(email, '@', 2) = 'spice.ai';"
HTTP/1.1 200 OK
content-type: application/json
x-cache: Hit from spiceai
results-cache-status: HIT
vary: Spice-Cache-Key
vary: origin, access-control-request-method, access-control-request-headers
content-length: 119
date: Thu, 24 Jul 2025 14:18:00 GMT
```

:::warning[Note]

> When supplying your own cache key, **you are responsible for asserting the semantic equivalence of your queries**. For example, this is expected behavior:

Reviewer: ```suggestion
When supplying a custom cache key, **ensure the semantic equivalence of queries**. For example, this is expected behavior:
```
